default_gateway.go

```diff
@@ -119,15 +130,15 @@ func (sb *sandbox) needDefaultGW() bool {
 
 func (sb *sandbox) getEndpointInGWNetwork() *endpoint {
 	for _, ep := range sb.getConnectedEndpoints() {
-		if ep.getNetwork().name == libnGWNetwork {
+		if ep.getNetwork().name == libnGWNetwork && strings.HasPrefix(ep.Name(), "gateway_") {
```

Comment:
I believe @msabansal is planning to modify the docker code as well if this change goes in as is.
I was hoping this function need not to be touched. This is pretty much driver independent.